src/TheThingsNetwork.cpp

```diff
@@ -396,19 +402,51 @@ void TheThingsNetwork::autoBaud()
   while (attempts-- && length == 0)
   {
     delay(100);
+    // Send break + Autobaud 
     modemStream->write((byte)0x00);
     modemStream->write(0x55);
     modemStream->write(SEND_MSG);
-    sendCommand(SYS_TABLE, 0, true, false);
-    sendCommand(SYS_TABLE, SYS_GET, true, false);
-    sendCommand(SYS_TABLE, SYS_GET_VER, false, false);
-    modemStream->write(SEND_MSG);
-    length = modemStream->readBytesUntil('\n', buffer, sizeof(buffer));
+    // check we can talk
+    length = checkModuleAvailable();
+    
+    // We succeeded talking to the module ?
+    baudDetermined = (length > 0) ;
   }
   delay(100);
   clearReadBuffer();
   modemStream->setTimeout(10000);
-  baudDetermined = true;
```

Comment:
I actually don't know why we added this flag to start with. Presumably to do a check before every `sendCommand` call and `autoBaud` if we didn't do so yet. If we do decide to implement this, we need to write the `autoBaud` and `checkModuleAvailable` functions not to use `sendCommand`.

But for now maybe let's just scrap the `baudDetermined` flag and rather introduce a `moduleSleeping` flag.

Reply:
Good point. Maybe not a flag then but an enum. Something like this perhaps? Please suggest naming.
```
enum state {
  UNKNOWN,
  AWAKE,
  SLEEPING
}
```

Reply:
sounds good, may be UNKNOWN => NOT_FOUND or NONE or MISSING